core/templates/dev/head/domain/topic_viewer/TopicViewerBackendApiServiceSpec.js

```diff
@@ -0,0 +1,72 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for TopicViewerBackendApiService.
+ */
+
+describe('Topic viewer backend API service', function() {
+  var TopicViewerBackendApiService = null;
+  var sampleDataResults = null;
+  var $rootScope = null;
+  var $scope = null;
+  var $httpBackend = null;
+  var UndoRedoService = null;
+
+  beforeEach(module('oppia'));
+
+  beforeEach(inject(function($injector) {
+    TopicViewerBackendApiService = $injector.get(
+      'TopicViewerBackendApiService');
+    $rootScope = $injector.get('$rootScope');
+    $scope = $rootScope.$new();
+    $httpBackend = $injector.get('$httpBackend');
+
+    // Sample topic object returnable from the backend
+    sampleDataResults = {
+      topic_name: 'topic_name',
+      canonical_story_dicts: {
+        id: '0',
+        title: 'Story Title',
+        description: 'Story Description',
+      },
+      additional_story_dicts: {
+        id: '0',
```

Comment:
Have different IDs for canonical and additional stories.

Reply:
Okay